### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Vova-Stoliar**

### CODE REVIEW

File: src/robotFactory.js [hunk 1]
```diff
@@ -1,10 +1,79 @@
...
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    if (Object.keys(coords).length === 0) {
+      this.coords = {
+        x: 0,
+        y: 0,
+      };
+    } else {
+      this.coords = coords;
+    }
```
Comment (DmytroMykoliv) [inline src/robotFactory.js:12]: Here you can simplified:
    this.coords = {
      x: 0,
      y: 0,
      ...coords,
    };
    without if / else  ))

**Review by DmytroMykoliv — APPROVED** [review]
Looks good ))

File: src/robotFactory.js (hunk 1 above)
Comment (Vova-Stoliar) [reply src/robotFactory.js]: genius)

File: src/robotFactory.js [hunk 2]
```diff
@@ -1,10 +1,76 @@
...
+  goLeft(steps = 1) {
+    this.coords.x -= steps;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    if (coords.z === undefined) {
+      coords.z = 0;
+    }
```
Comment (VinnieJ-2k20) [inline src/robotFactory.js:44]: Simpler way to write this:
```suggestion
    this.coords.z = coords.z || 0;
```

**Review by VinnieJ-2k20 — APPROVED** [review]
Good job 👍 
Added one comment.

**Pull request closed without merging**